Gems/ROS2/Code/Source/Camera/CameraSensorConfiguration.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzCore/RTTI/RTTI.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/std/string/string.h>
+#include <ROS2/Camera/ImagePostprocessingFunction.h>
+
+namespace ROS2
+{
+    //! A structure capturing configuration of a single camera sensor with up to two image sources (color and depth).
+    struct CameraSensorConfiguration
+    {
+        AZ_TYPE_INFO(CameraSensorConfiguration, "{386A2640-442B-473D-BC2A-665D049D7EF5}");
+        static void Reflect(AZ::ReflectContext* context);
+
+        float verticalFieldOfViewDeg = 90.0f; //!< vertical field of view of camera sensor.
+        int width = 640; //!< Camera image width in pixels.
+        int height = 480; //!< Camera image height in pixels.
+        bool colorCamera = true; //!< Use color camera?
+        bool depthCamera = true; //!< Use depth camera?
```

Comment:
standards: Missing `m_` prefix (this is usually expected for all class members, even if public)

(ignore me if the standard O3DE coding standards don't apply here)